src/pretix/base/timeframes.py

```diff
@@ -0,0 +1,404 @@
+        lambda ref_d: ref_d.replace(day=1, month=1, year=ref_d.year + 1),
+        lambda ref_d, start_d: start_d.replace(day=31, month=12),
+        True,
+        pgettext_lazy('reporting_timeframe', 'by year'),
+        lambda start_d, end_d: str(start_d.year),
+    ),
+    (
+        'future',
+        pgettext_lazy('reporting_timeframe', 'All future (excluding today)'),
+        lambda ref_d: ref_d + timedelta(days=1),
+        lambda ref_d, start_d: None,
+        True,
+        pgettext_lazy('reporting_timeframe', 'Other'),
+        lambda start_d, end_d: date_format(start_d, "SHORT_DATE_FORMAT") + ' – ',
+    ),
+    (
+        'past',
+        pgettext_lazy('reporting_timeframe', 'All past (including today)'),
+        lambda ref_d: None,
+        lambda ref_d, start_d: ref_d,
+        True,  # technically false, but only makes sense to have in a selection that also allows the future, otherwise redundant
+        pgettext_lazy('reporting_timeframe', 'Other'),
+        lambda start_d, end_d: ' – ' + date_format(end_d, "SHORT_DATE_FORMAT"),
+    ),
+)
+
+
+class DateFrameWidget(forms.MultiWidget):
+    template_name = 'pretixbase/forms/widgets/dateframe.html'
+
+    def __init__(self, *args, **kwargs):
+        self.timeframe_choices = kwargs.pop('timeframe_choices')
+        widgets = (
+            forms.Select(choices=self.timeframe_choices),
+            forms.DateInput(attrs={'class': 'datepickerfield', 'placeholder': pgettext_lazy('timeframe', 'Start')}),
+            forms.DateInput(attrs={'class': 'datepickerfield', 'placeholder': pgettext_lazy('timeframe', 'End')}),
+        )
+        super().__init__(widgets=widgets, *args, **kwargs)
+
+    def decompress(self, value):
+        if not value:
+            return ['unset', None, None]
+        if '/' in value:
+            return [
+                'custom',
+                date.fromisoformat(value.split('/', 1)[0]),
+                date.fromisoformat(value.split('/', 1)[-1]),
+            ]
+        return []
+
+    def get_context(self, name, value, attrs):
+        ctx = super().get_context(name, value, attrs)
+        ctx['required'] = self.timeframe_choices[0][0] == 'unset'
+        return ctx
+
+
+def _describe_timeframe(label, start, end, future, describe):
+    d_start = start(now())
+    d_end = end(now(), d_start)
+    details = describe(d_start, d_end)
+    return f'{label} ({details})'
+
+
+class DateFrameField(forms.MultiValueField):
+    default_error_messages = {
+        **forms.MultiValueField.default_error_messages,
+        'inconsistent': gettext_lazy('The end date must be after the start date.'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        include_future_frames = kwargs.pop('include_future_frames')
+
+        top_choices = [('custom', gettext_lazy('Custom timeframe'))]
+        if not kwargs.get('required', True):
+            top_choices.insert(0, ('unset', pgettext_lazy('reporting_timeframe', 'All time')))
```

Comment:
Not super keen on the wording „All time“. I do not know something better though, maybe „no limit“ or „do not limit“ or just „–––“ or maybe „None“?

Reply:
Hm, no, I disagree.

"––" or "None" might make the user think that they need to select something, but they don't, it's a valid option to keep empty.

"Date range: [nothing]" also feels less intuitive than "Date range: all time" to me.

Reply:
Okay, I agree!